library/src/webapp/js/jquery/qtip/tutorial.js

```diff
@@ -5,129 +5,145 @@ var optsCache;
+                        button: $('<a class="qtipClose tut-close" href="#" onclick="if(\''+opts.showTutorialLocationOnHide + '\' == \'true\' && \'' + url + '\' != \'' + sakaiTutorialLocationUrl + '\'){showTutorialPage(\''+ sakaiTutorialLocationUrl + '\');}" title="' + $('.closeMe').find('.skip').text() +'"><i class="fa fa-close tut-icon-close"></i><span class="skip">' + $('.closeMe').find('.skip').text() + '</span></a>'),
+                        text: response.data.body
+                    },
+                    position: response.data.dialog == 'true' ? dialogPosition: {
+                        my: response.data.positionTooltip,
+                        at: response.data.positionTarget,
+                        viewport: $(document.body),
+                        adjust: { method: 'shift' }
+                    },
+                    style: {
+                        classes: 'sakai-tutorial qtip-shadow',
+                        tip: {
+                            corner: response.data.positionTooltip
+                        }
+                    },
+                    show: {
+                        ready: true,
+                        solo: true
+                    },
+                    hide: false,
+                    events: {
+                        hide: function(event, api)
+                        {
+                            $(response.data.selection).qtip("destroy");
+                            $('#skipNav').attr('tabindex','-1').focus();
+                            endTutorial(response.data.selection);
+                        },
+                        show: function(event, api)
+                        {
+                            $('.qtipClose, .tut-close').on('click', function(e) {
+                                api.hide(event);
+                                endTutorial(response.data.selection);
+                            });
+                            if(response.data.fadeout){
+                                setTimeout(function(){
+                                    $('.qtip').fadeOut(2000, function() {
+                                        $(response.data.selection).qtip("destroy");
+                                    });
+                                }, 10000);
+                            }
+                        },
+                        visible : function() {  $('.qtip-title').attr('tabindex','-1').focus(); },
+                        render: function(event, api) {
+                            $(window).bind('keydown', function(e) {
+                                if(e.keyCode === 27) {
+                                    api.hide(event);
+                                    $(response.data.selection).qtip("destroy");
+                                }
+                            });
+                        }
+                    }
+                });
+            }
+        } catch(e) {
+            // Handle exception here
```

Comment:
would be ideal to not leave a catch block empty, including a log message minimally would be helpful

Reply:
I forgot to catch :), will fix it now :)